### ISSUE TITLE
[BUG] - Refresh required after delete/move/rename folder via terminal

### ISSUE THREAD

**Issue by kcpevey** [nebari-dev/jupyterlab-gallery#46]
### Feature description

I deleted the folder where my gallery was supposed to be downloaded to. Now the UI won't let me fetch the latest changes. I think it thinks that I still have the latest? I added the folder back and I still can't fetch. 

<img width="285" alt="image" src="https://github.com/user-attachments/assets/fdb9b3e0-ebe3-4a4f-b777-eaad25689b58">

### Value and/or benefit

Users should be able to recover from deleting the gallery folder.

**Comment by krassowski** [comment]
Which folder? The one with `.git` or parent?

**Comment by kcpevey** [comment]
I deleted `examples`. Ah, I just realized from the error message when I click the folder icon that it's looking for `examples/playground-examples`. I recreated that folder, but I'm still in the same stuck situation. I can't pull the repo.

**Comment by kcpevey** [comment]
Its not clear from the plugin, what's going where. Based on the error message when I clicked on the folder icon - I'd say `playground-examples` is the .git folder?

**Comment by krassowski** [comment]
Can you delete `playground-examples` (but keep `examples`), and reload JupyterLab?

**Comment by kcpevey** [comment]
That worked.

I deleted `playground-examples`, but kept `examples`, reloaded jupyterlab. The plugin works again. 

Also, I deleted `examples` entirely, reloaded Jupyterlab and the plugin worked again. 

The reload was key.

**Comment by krassowski** [comment]
Normally it should refresh the status on delete:

https://github.com/nebari-dev/jupyterlab-gallery/blob/9de8f2cd02e5ef863abf7c02bc02d11e3ed4f02f/src/gallery.tsx#L70-L75

unless you renamed the folder (https://github.com/nebari-dev/jupyterlab-gallery/issues/29) or deleted/renamed/moved it from terminal, in which case it would update the status once the folder listing is refreshed (I think every 10 seconds?)

**Comment by kcpevey** [comment]
I deleted from the terminal. Deleting through the JLab UI would have required the painful process of deleting all the contents of the folder, plus viewing hidden files and deleting those as well. 

So we have no way to catch deleted/renamed/moved from terminal without building a more complicated watcher?

**Comment by krassowski** [comment]
There is pretty advanced file watcher logic in jupyter-server (https://github.com/jupyter-server/jupyter_server_fileid), but deleting from terminal circumvents it.

Two solutions:
- we can have a custom watcher
- we can allow removing folder with the contents; of note this is not a jupyter restriction, locally for me folders are always deleted; the relevant configuration options are `FileContentsManager.always_delete_dir` and `FileContentsManager.delete_to_trash`; there is also an interaction with the filesystem used

**Comment by kcpevey** [comment]
I think you'll always have _some_ people who operate from the terminal and find themselves in this situation. 

Another solution would be an info bubble on the gallery card with a popup modal that warned me a refresh was required if the folder is deleted.

**Comment by krassowski** [comment]
> an info bubble on the gallery card with a popup modal that warned me a refresh was required if the folder is deleted

If we already knew that we need to show the bubble, we could have refreshed the model in the background, eliminating the need for showing the bubble, right?

**Comment by krassowski** [comment]
> I think you'll always have some people who operate from the terminal and find themselves in this situation.

Absolutely! Maybe the solution is along what you proposed, but instead triggering a forced model update on mouse hover if it has not been updated in last 3 seconds or so. This might be cheaper than having an always running file watcher.

**Comment by kcpevey** [comment]
>If we already knew that we need to show the bubble, we could have refreshed the model in the background, eliminating the need for showing the bubble, right?

I was not thinking about a dynamic bubble. I was thinking of a static info bubble on all gallery cards at all times - what is this tile, what does it do, and also, note that if you delete the folder, you'll have to refresh the page.

**Comment by krassowski** [comment]
There is an issue on `jupyter_server_fileid` now about potentially making it watch the filesystem changes performed outside of jupyter: https://github.com/jupyter-server/jupyter_server_fileid/issues/80
